pkg/cmd/experimental/syncgroups/grouppruner_test.go

```diff
@@ -0,0 +1,106 @@
+package syncgroups
+
+import (
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/client/testclient"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+)
+
+func TestGoodPrune(t *testing.T) {
+	testGroupPruner, tc := newTestPruner()
+	errs := testGroupPruner.Prune()
+	for _, err := range errs {
+		t.Errorf("unexpected prune error: %v", err)
+	}
+
+	checkClientForDeletedGroups(tc, []string{"os" + Group2UID}, t)
+}
+
+func TestListFailsForPrune(t *testing.T) {
```

Comment:
There should be tests to be sure existence checks for one group don't fail the whole job